docs/dev/design_documents/data_collector.md

```diff
@@ -0,0 +1,567 @@
+---
+title: Data Collector
```

Comment:
In my review of this PR, I stopped halfway through this file. Will pick up my review here for part 2.